### ISSUE TITLE
Rename and consolidate binaries and container images

### ISSUE THREAD

**Issue by mauriciovasquezbernal** [inspektor-gadget/inspektor-gadget#2100]
TL;DR: We have too many binaries and container images, it's difficult to understand for the users and to handle for us.

Inspektor Gadget currently provides two main tools:

- Inspektor Gadget on Kubernetes, a.k.a [gadgettracermanager](https://github.com/inspektor-gadget/inspektor-gadget/blob/v0.21.0/gadget-container/gadgettracermanager/main.go): A tool to trace and inspect applications running in Kubernetes. It was the first tool we created as the initial scope of the project was only about Kubernetes.
- [`ig`](https://github.com/inspektor-gadget/inspektor-gadget/blob/v0.21.0/docs/ig.md): A tool to debug processes and applications on the local host. 

Each of these tools have a client part, [kubectl-gadget](https://github.com/inspektor-gadget/inspektor-gadget/tree/v0.21.0/cmd/kubectl-gadget) and [gadgetctl](https://github.com/inspektor-gadget/inspektor-gadget/tree/v0.21.0/cmd/gadgetctl) and there is a container image published for each of them [inspektor-gadget](https://github.com/inspektor-gadget/inspektor-gadget/pkgs/container/inspektor-gadget) & [ig](https://github.com/inspektor-gadget/inspektor-gadget/pkgs/container/ig).

The gadgettracermanager and ig binaries were too different at the beginning, but after much refactoring these two two binaries are becoming more and more similar on its implementation and use cases and we're considering adding features present in one to the other, for instance, #737, #1861 , #1862. 

The main differences between the two binaries right now are:
- gadgettracermanager has some kubernetes specific logic:
  - contacts the API server to 
    - get notifications about pod creation
    - list pods and services
    - implement the tracer controller 
  - use specific operators for Kubernetes (KubeIPResolver)

Regarding the container images, the inspektor-gadget one has some additional things like:
- entrypoint.sh and cleanup.sh
- hooks 

The differences are not that big, so the proposal of this issue is to consolidate them in a single binary and container image. In order to enable the Kubernetes specific features, we could have a `gadget mode` env variable or command line argument. One important thing is to be sure the UX is clean, for instance a user using `ig` on a local node shouldn't have all Kubernetes related flags available, unless explicitly requested. 

Regarding the naming, it should be called `ig` and the container image [inspektor-gadget](https://github.com/inspektor-gadget/inspektor-gadget/pkgs/container/inspektor-gadget) .

It seems to me it's also possible to consolidate gadgetctl and kubectl-gadget on the same binary, but it needs more thinking.

**Comment by blixtra** [comment]
Sounds good to me!
